### PR TITLE
Refactor everything!

### PR DESCRIPTION
**Pull request by kashike**
🥞

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,9 +9,9 @@ jobs:
       - name: "checkout repository"
         uses: "actions/checkout@v2"
       - name: "setup java"
-        uses: actions/setup-java@v2
+        uses: "actions/setup-java@v2"
         with:
-          distribution: 'adopt'
-          java-version: '17'
+          distribution: "adopt"
```
Comment (Proximyst) [inline .github/workflows/build.yml:14]: Should be `temurin` (or another distribution) as adopt is no longer updated.

Comment (kashike) [reply .github/workflows/build.yml]: https://github.com/PaperMC/bibliothek/pull/76/commits/b286b1e95fefacb1b817312bb982099feac98011

File: src/main/java/io/papermc/bibliothek/controller/v2/VersionFamilyController.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  @ApiResponse(
+    content = @Content(
+      schema = @Schema(implementation = VersionFamilyResponse.class)
+    ),
+    responseCode = "200"
+  )
+  @GetMapping("/v2/projects/{project:[a-z]+}/version_group/{versionGroup:[0-9pre.-]+}")
+  @Operation(summary = "Get information about a version group.")
```
Comment (Aberdeener) [inline src/main/java/io/papermc/bibliothek/controller/v2/VersionFamilyController.java:77]: "Get information about a ***project's*** version group" might read better here?

File: src/main/java/io/papermc/bibliothek/controller/v2/VersionFamilyBuildsController.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  }
+
+  @ApiResponse(
+    content = @Content(
+      schema = @Schema(implementation = VersionFamilyBuildsResponse.class)
+    ),
+    responseCode = "200"
+  )
+  @GetMapping("/v2/projects/{project:[a-z]+}/version_group/{versionGroup:[0-9pre.-]+}/builds")
+  @Operation(summary = "Get all available builds for a version group.")
```
Comment (Aberdeener) [inline src/main/java/io/papermc/bibliothek/controller/v2/VersionFamilyBuildsController.java:85]: Same thing, maybe add the word "project's" here

**Review by Aberdeener — APPROVED** [review]

File: src/main/java/io/papermc/bibliothek/controller/v2/ProjectsController.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  @ApiResponse(
+    content = @Content(
+      schema = @Schema(implementation = ProjectsResponse.class)
+    ),
+    responseCode = "200"
+  )
+  @GetMapping("/v2/projects")
+  @Operation(summary = "Gets a list of all available projects.")
```
Comment (NoahvdAa) [inline src/main/java/io/papermc/bibliothek/controller/v2/ProjectsController.java:63]: Super tiny nitpick here, you use `Get ...` everywhere else.
```suggestion
  @Operation(summary = "Get a list of all available projects.")
```

**Pull request merged**